docs/editor-neovim.qmd

```diff
@@ -0,0 +1,63 @@
+---
+title: "Neovim"
+editor:
+  markdown:
+    wrap: sentence
+    canonical: true
+---
+
+# Installing the air client
+
+First, install the Air client by following the instructions on the [Air CLI documentation](https://posit-dev.github.io/air/cli.html).
+
+# Configuration
+
+Air can be configured as a Language Server Protocol (LSP) client via [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) or as a formatter plugin via [conform.nvim](https://github.com/stevearc/conform.nvim).
+
+## LSP Configuration with nvim-lspconfig
+
+To configure Air as an LSP client, add the following to your `nvim/lua/plugins/lspconfig.lua`:
+
+```lua
+require("lspconfig").air.setup({})
+```
+
+To enable automatic formatting on save, add the following to your `nvim/after/plugin/lsp.lua` (or other appropriate location):
+
+```lua
+vim.api.nvim_create_autocmd("BufWritePre", {
+    pattern = { "*.R", "*.r" },
+    callback = function()
+        vim.lsp.buf.format({
+            async = true,
+            filter = function(client)
+                return client.name == "air"
+            end,
+        })
+    end,
+})
+```
+
+## conform.nvim configuration
+
+Air can be configured as a formatter plugin via [conform.nvim](https://github.com/stevearc/conform.nvim). This allows you to format R code with Air inside code chunks in Quarto and RMarkdown documents. It can be configured by adding the following to your `nvim/lua/plugins/conform.lua`:
+
+```lua
+return {
+    "stevearc/conform.nvim",
+    enabled = true,
+    config = function()
+        local conform = require("conform")
+
+        conform.setup({
+            notify_on_error = true,
+            format_on_save = {
+                lsp_fallback = true,
+            },
+            formatters_by_ft = {
+                r = { "air" },
+            },
+        })
+    end,
+}
```

Comment:
That's a `lazy.nvim`-specific setup. I'd suggest using a generic setup (example below) and point to conform's README for a way to install the plugin.

```lua
require("conform").setup({
  formatters_by_ft = {
    r = { "air" },
  },
})
```

What about filetypes for R markdown and Quatro?

Reply:
Currently not supported by air.